Cargo.toml

```diff
@@ -1,49 +1,64 @@
-[package]
-name = "async-compression"
-version = "0.4.18"
+[workspace]
+members = ["crates/*"]
+resolver = "2"
+
+[workspace.package]
 authors = ["Wim Looman <wim@nemo157.com>", "Allen Bui <fairingrey@gmail.com>"]
 edition = "2018"
 license = "MIT OR Apache-2.0"
-keywords = ["compression", "gzip", "zstd", "brotli", "async"]
-categories = ["compression", "asynchronous"]
 repository = "https://github.com/Nullus157/async-compression"
+
+[workspace.dependencies]
+compression-codecs = { version = "0.1.0", path = "crates/compression-codecs" }
+compression-core = { version = "0.1.0", path = "crates/compression-core" }
+
+[package]
+name = "async-compression"
+version = "0.4.18"
 description = """
 Adaptors between compression crates and Rust's modern asynchronous IO types.
 """
+authors.workspace = true
+edition.workspace = true
+license.workspace = true
+repository.workspace = true
+keywords = ["compression", "gzip", "zstd", "brotli", "async"]
+categories = ["compression", "asynchronous"]
 
 [package.metadata.docs.rs]
 all-features = true
 rustdoc-args = ["--cfg", "docsrs"]
 
 [features]
+default = ["all"]
+
 # groups
 all = ["all-implementations", "all-algorithms"]
 all-implementations = ["futures-io", "tokio"]
 all-algorithms = ["brotli", "bzip2", "deflate", "gzip", "lzma", "xz", "zlib", "zstd", "deflate64"]
 
 # algorithms
-deflate = ["flate2"]
-gzip = ["flate2"]
-lzma = ["xz2"]
-xz = ["xz2"]
-zlib = ["flate2"]
-zstd = ["libzstd", "zstd-safe"]
-zstdmt = ["zstd", "zstd-safe/zstdmt"]
-deflate64 = ["dep:deflate64"]
+brotli = ["compression-codecs/brotli"]
+bzip2 = ["compression-codecs/bzip2"]
+deflate = ["compression-codecs/deflate", "flate2"]
+deflate64 = ["compression-codecs/deflate64"]
+flate2 = ["compression-codecs/flate2"]
+gzip = ["compression-codecs/gzip", "flate2"]
+lzma = ["compression-codecs/lzma", "xz2"]
+xz = ["compression-codecs/xz", "xz2"]
+xz2 = ["compression-codecs/xz2"]
+zlib = ["compression-codecs/zlib", "flate2"]
+zstd = ["compression-codecs/zstd"]
+zstdmt = ["compression-codecs/zstdmt"]
+libzstd = ["compression-codecs/libzstd"]
+zstd-safe = ["compression-codecs/zstd-safe"]
```

Comment:
```suggestion
```

I don't think we should forward this two features, which is enabled when zstd is enabled

Reply:
Technically, not having these features is a breaking change as we currently have them. I kept them for this reason. I would suggest keeping them and removing them with the next bump of the minor version.

Reply:
Thanks I didn't realize that it's implicitly added due to not using `deps:`

Reply:
Thanks I didn't realize that it's implicitly added due to not using `deps:`